src/pydicom/pixels/utils.py

```diff
@@ -673,11 +694,18 @@ def decompress(
 
     # Pad with 0x00 if odd length
     nr_frames = len(frames)
-    if value_length % 2:
-        frames.append(b"\x00")
-
     elem = ds["PixelData"]
-    elem.value = b"".join(frame for frame in frames)
+
+    if ds.BitsAllocated == 1:
+        elem.value = concatenate_packed_frames(
+            frames, frame_length=ds.Rows * ds.Columns
+        )
+    else:
+        if value_length % 2:
+            frames.append(b"\x00")
+
+        elem.value = b"".join(frame for frame in frames)
```

Comment:
Do you mind changing this to `elem.value = b"".join(frames)`? Overlooked it in the original